node/src/components/storage/disjoint_sequences.rs

```diff
@@ -437,6 +457,94 @@ mod tests {
         );
     }
 
+    #[test]
+    fn should_truncate() {
+        const SEQ_HIGH: Sequence = Sequence { high: 11, low: 9 };
+        const SEQ_MID: Sequence = Sequence { high: 6, low: 6 };
+        const SEQ_LOW: Sequence = Sequence { high: 3, low: 1 };
+        fn new_sequences() -> DisjointSequences {
+            DisjointSequences {
+                sequences: vec![SEQ_HIGH, SEQ_MID, SEQ_LOW],
+            }
+        }
```

Comment:
Can't you initialize it once and then only clone it?

Reply:
`DisjointSequences` isn't clonable.  I can add that behind a `test` feature-gate if you have a strong preference, but I'm not sure it's worth it?

Reply:
It's not a blocker but it _could_ be clonable since it's just a couple tuples.

Reply:
Fair enough - done in [c13fba6](https://github.com/casper-network/casper-node/pull/3191/commits/c13fba604943a73ec0642a72f712e3a433d75525).